src/pages/YearPickerPage.js

```diff
@@ -102,20 +92,17 @@ class YearPickerPage extends React.Component {
                     title={this.props.translate('yearPickerPage.year')}
                     onBackButtonPress={() => Navigation.goBack(`${this.props.route.params.backTo}?year=${this.currentYear}` || ROUTES.HOME)}
                 />
-                <OptionsSelector
+
+                <SelectionListRadio
                     textInputLabel={this.props.translate('yearPickerPage.selectYear')}
+                    textInputValue={this.state.inputText}
+                    textInputMaxLength={4}
                     onChangeText={this.filterYearList}
                     keyboardType={CONST.KEYBOARD_TYPE.NUMBER_PAD}
-                    maxLength={4}
-                    value={this.state.inputText}
+                    headerMessage={headerMessage}
                     sections={[{data: this.state.yearOptions, indexOffset: 0}]}
```

Comment:
SelectionListRadio uses `sections` as a dependency for `useMemo` , here you are creating a new array on every render, which will execute the `useMemo` as it is a new object.
Objects `{}` and Arrays `[]` aren't stable types, I fixed a [bug related to this](https://github.com/Expensify/App/issues/19700#issuecomment-1591230222). 
Or you memo the `sections` like

```
const sections = useMemo(() => [{data: this.state.yearOptions, indexOffset: 0}], [this.state.yearOptions] );
```

or if doesn't worth it because all will change on render, the `useMemo` in `SelectionListRadio` isn't needed

Reply:
@gedu You're definitely right, that `useMemo` was a leftover and is not needed. Thanks!